src/base/system.cpp

```diff
@@ -266,6 +266,57 @@ unsigned io_read(IOHANDLE io, void *buffer, unsigned size)
 	return fread(buffer, 1, size, (FILE *)io);
 }
 
+void io_read_all(IOHANDLE io, void **result, unsigned *result_len)
+{
+	int signedLen = io_length(io);
```

Comment:
```suggestion
	int signed_len = io_length(io);
```
Or you could do away with the `signed_len`, you can also compare the unsigned length with `(unsigned)-1`.

Reply:
I changed the name. Even if it's not technically undefined behavior to do `(unsigned)-1`, I don't really like the looks of it :)